### PR TITLE
fix: mypy [no-untyped-def] typing (TDE-264)

### PR DESCRIPTION
**Pull request by amfage**
**Change Description:**
Add missing typing in functions, mypy error code `no-untyped-def`
[https://linzgovt.atlassian.net/browse/TDE-264](https://linzgovt.atlassian.net/browse/TDE-264)

**Notes for Testing:**
run `mypy` . should not display any error.

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import pytest
 
 
-def pytest_addoption(parser):
+def pytest_addoption(parser) -> None:  # type: ignore
```
Comment (paulfouquet) [inline conftest.py:4]: The `ignore` comment was for the function parameter `parser`?

File: pyproject.toml [hunk 2]
```diff
@@ -48,7 +48,7 @@ show_error_codes = true
 strict = true
 disable_error_code = [
     "attr-defined",
-    "no-untyped-def",
+#    "no-untyped-def",
```
Comment (paulfouquet) [inline pyproject.toml:51]: I suggest to remove it rather than commenting the line

File: topo_processor/cog/command.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import os
-from typing import List, Optional, TypedDict
+from typing import List, Literal, Optional, Tuple, TypedDict
```
Comment (paulfouquet) [inline topo_processor/cog/command.py:2]: Is `Literal` used in this file?

File: topo_processor/cog/execution.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import subprocess
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Literal, Tuple
```
Comment (paulfouquet) [inline topo_processor/cog/execution.py:2]: Same here for `Literal`

File: topo_processor/file_system/get_fs.py [hunk 5]
```diff
@@ -1,6 +1,10 @@
+from typing import Any
+
 from fsspec.implementations.local import LocalFileSystem
 from s3fs import S3FileSystem
 
+from topo_processor import file_system
```
Comment (paulfouquet) [inline topo_processor/file_system/get_fs.py:6]: Is it an unused import?

File: topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py [hunk 6]
```diff
@@ -61,7 +61,7 @@ def load_metadata(self, asset: Asset = None, metadata_file: str = "", is_load_al
             asset.key_name = AssetKey.Visual
             self.populate_item(asset_metadata, asset)
 
-    def populate_item(self, metadata_row, asset: Asset = None) -> None:
+    def populate_item(self, metadata_row: Dict[str, str], asset: Asset = None) -> None:
```
Comment (paulfouquet) [inline topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py:64]: Is `metadata_row` always a `Dict[str, str]`? Can it be a Dict[str, int]` or `Dict[str, float]` too?

File: topo_processor/util/conversions.py [hunk 7]
```diff
@@ -24,7 +24,7 @@ def remove_empty_strings(properties: Dict) -> Dict:
     return {key: value for key, value in properties.items() if value != ""}
 
 
-def string_to_boolean(value, true_values, false_values):
+def string_to_boolean(value: str, true_values: list, false_values: list) -> Union[bool, str]:
```
Comment (paulfouquet) [inline topo_processor/util/conversions.py:27]: They might be a `list` of `str`. You want to use `List[str]` in this case. (`List` is the annotation type for `list`)

**Review by paulfouquet — CHANGES_REQUESTED** [review]
The changes look good :+1: 
Just a couple of suggestions and few unused import to fix.

File: topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py (hunk 6 above)
Comment (amfage) [reply topo_processor/metadata/metadata_loaders/metadata_loader_imagery_historic.py:64]: Just noting we took a look and it is Dict[str, str] but good to check it.

**Review by MDavidson17 — APPROVED** [review]
:+1:

File: topo_processor/util/valid.py [hunk 8]
```diff
@@ -1,14 +1,18 @@
+from getopt import error
```
Comment (paulfouquet) [inline topo_processor/util/valid.py:1]: I think they are unused import

**Review by paulfouquet — CHANGES_REQUESTED** [review]
A couple of unused import left

**Review by paulfouquet — APPROVED** [review]
:+1:

**Pull request merged**